packages/qwik/src/optimizer/src/plugins/plugin.ts

```diff
@@ -875,8 +875,38 @@ export const manifest = ${JSON.stringify(manifest)};\n`;
   // order by discovery time, so that related segments are more likely to group together
   function manualChunks(id: string, { getModuleInfo }: Rollup.ManualChunkMeta) {
     const module = getModuleInfo(id)!;
-    const segment = module.meta.segment as SegmentAnalysis | undefined;
-    return segment?.entry;
+    const segment = module.meta.segment;
```

Comment:
you should still return entry if segment exists, that's how insights and entry strategy do their thing